Bits AI docs (#19695)
DataDog/documentation: content/en/bits_ai/_index.md

- Who is on call for example-service?
- Find me the example-service dashboard.

Bits AI also expands on issues with upstream and downstream dependencies. This feature works best if your APM services are tagged by **team** and **service**.

For details, see [Getting Started][5] and [Query Examples][3].

### Streamline incident management

As part of the [Datadog Incident Management][2] product, generative AI can help you 

- **Stay informed:** Ask Datadog to send you an update on the incident in 15 minutes.
- **Explore incidents:** Query incident details, such as "Give me a few bullet points on incident 2781."
- **Triage incidents:** Update the severity level and status of an incident.
- **Generate a postmortem:** Quickly create a first draft to review and revise.

For details, see [Managing Incidents][4].

### Run workflows in Slack

If you have Datadog Workflows configured, you can trigger them from Slack. For example, sending `@Datadog Give me a workflow to block IPs in Cloudflare` starts a conversation in which the chatbot confirms the desired workflow and asks for any required parameters.

<span style='background-color: yellow'>[SCREENSHOT OF SLACK WORKFLOWS]</span>

> ## Privacy

Author: Will delete this section, per discussion with Kai.

---
title: Bits AI
kind: documentation
disable_toc: false
private: true
is_beta: true
further_reading:
    - link: "https://www.datadoghq.com/product/platform/bits-ai/"
      tag: "Product page"
      text: "Bits AI"
    - link: "https://www.datadoghq.com/blog/datadog-bits-generative-ai/"
      tag: "Blog post"
      text: "Introducing Bits AI, Your New DevOps Copilot"
    - link: "https://www.datadoghq.com/about/latest-news/press-releases/datadog-announces-bits-an-ai-assistant-to-help-engineers-quickly-resolve-application-issues/"
      tag: "Press release"
      text: "Bits AI announcement"
---

> <div class="alert alert-info">Bits AI is in private beta.</div>

Reviewer: Fill out this [form](dashcon.io/bits-ai) to join the waitlist.

Author: @kaixintai That link doesn't seem to be working for me, can you try again?

> Bits AI is a generative AI interface that helps you identify, remediate, and summarize issues in your applications and infrastructure. It surfaces faulty deployments, Watchdog anomalies, incidents, alerts, and more.

Reviewer: ```suggestion
Bits AI is a generative AI interface that helps you identify, remediate, and summarize issues in your applications and infrastructure. It surfaces faulty deployments, Watchdog anomalies, incidents, alerts, and more.
```

> Bits AI is a generative AI interface that helps you identify, remediate, and summarize issues in your applications and infrastructure. It surfaces faulty deployments, Watchdog anomalies, incidents, alerts, and more.

Reviewer: I also recommend moving this sentence "It surfaces faulty deployments, Watchdog anomalies, incidents, alerts, and more." to the [Query data in natural language](https://docs-staging.datadoghq.com/jen.gilbert/DOCS-6036-bits-ai/bits_ai/#query-data-in-natural-language) section. See my comment in that section.

> ## Features

Author: Generally we want to explain what the feature is before we explain the steps for using it, but I added a link closer to the top of the page (and before the features section) for folks who are eager to start.

I didn't include Getting Started in this page content because it makes the page headers more nested, which means the content is harder to navigate, since the righthand nav on the page will only show the top headers. It also just makes the page quite long, so prominent links are usually the way to go.

### Query data in natural language

Bits AI supports natural language querying for logs, APM traces, infrastructure data, and cloud cost.

> In Datadog or in Slack, you can ask Bits AI questions such as

Reviewer: You can also ask Bits AI to tell you about the health and ownership of your services, and retrieve Datadog resources related to those services.

- What is going on with example-service?
- Are there any issues with example-service's dependencies?
- Who is on call for example-service?
- Find me the example-service dashboard.

> Bits AI also expands on issues with upstream and downstream dependencies. This feature works best if your APM services are tagged by **team** and **service**.

Reviewer: I found this section a little difficult to follow. I'll try to describe why. Let me know if it makes sense:
1. When you ask the first two questions, "What is going on with example-service?" and "Are there any issues with example-service's dependencies", this sentence from the top of the page applies "it surfaces faulty deployments, Watchdog anomalies, incidents, alerts, and more.". In addition to telling you about the service you asked about, it will also expand on upstream/downstream services. 
2. Bits AI does not tell you about your dependencies when you ask "Who is on call for example-service?" and "Find me the example-service dashboard.". 

It was hard to map the sentence about how Bits answers expansively to the first two questions in the list.

Author: I didn't intend any particular relationship between the example queries and the sentences below them, the examples were just intended purely as natural language examples. But if it confused you, it will confuse someone else, so I've revised it! Thanks for flagging.

> {{< whatsnext desc="Additional documentation:">}}

Reviewer: Is it possible to include these links in line vs having them in a separate section like this? The getting started page is very important and I worry that it makes it difficult for our customers to find that info. I would prefer having the contents of that page in a prerequisites section on this main page.

Author: See my comments elsewhere for why the Getting Started content isn't inline above this -- but I added a link at the top of the page. I used a section like this because its nav style is sometimes easier for the reader to see than inline links, so I do recommend keeping it as well, but using both doesn't hurt!

    {{< nextlink href="bits_ai/getting_started/" >}}Getting Started{{< /nextlink >}}
    {{< nextlink href="bits_ai/query_examples/" >}}Example Natural Language Queries{{< /nextlink >}}
{{< /whatsnext >}}

### Streamline incident management

> As part of the [Datadog Incident Management][2] product, generative AI can help you 

Reviewer: ```suggestion
Within the [Datadog Incident Management][2] product, Bits AI can help you 
```

- **Stay informed:** Ask Datadog to send you an update on the incident in 15 minutes.

> - **Explore incidents:** Query incident details, such as "Give me a few bullet points on incident 2781."

Reviewer: This is the same as the bullet above so I think we can cut this

Reviewer: We can replace with: 
**Engage responders:** Page in on-call teams with PagerDuty. For example, "@Datadog Page the team that owns example-service"

> - **Triage incidents:** Update the severity level and status of an incident.

Reviewer: Triage is probably not the right word, maybe "update incidents"?

- **Generate a postmortem:** Quickly create a first draft to review and revise.

{{< whatsnext desc="Additional documentation:">}}
    {{< nextlink href="bits_ai/managing_incidents/" >}}Managing Incidents{{< /nextlink >}}
{{< /whatsnext >}}

### Run workflows in Slack

> If you have Datadog Workflows configured, you can trigger them from Slack. For example, sending `@Datadog Give me a workflow to block IPs in Cloudflare` starts a conversation in which the chatbot confirms the desired workflow and asks for any required parameters.

Reviewer: ```suggestion
If you have Datadog Workflows configured, you can trigger them from Slack. For example, sending `@Datadog Give me a workflow to block IPs in Cloudflare` starts a conversation in which Bits AI confirms the desired workflow and asks for any required parameters.
```

> {{< img src="bits_ai/slack-workflow.png" alt="Workflow request to Bits AI in Slack" style="width:60%;">}}

Reviewer: Could you use a screenshot of this thread instead? https://datadogincidents.slack.com/archives/C05J0SS2VFC/p1690391995008239?thread_ts=1690391890.822609&cid=C05J0SS2VFC

Author: That thread is pretty long for a screenshot, would the first 4 messages work? I can also break the conversation into a few screenshots if it's important to you to have the entire thing, just let me know.

Reviewer: Yeah, a couple messages starting from "Give me a workflow..." works

## Features

### Query data in natural language

Bits AI supports natural language querying for logs, APM traces, infrastructure data, and cloud cost. You can also ask Bits AI about the health and ownership of your services, and retrieve Datadog resources related to those services.

In Datadog or in Slack, you can ask Bits AI questions such as
- Who is on call for example-service?
- Find me the example-service dashboard.
- What is going on with example-service?
- Are there any issues with example-service's dependencies?

When relevant to your query, Bits AI surfaces faulty deployments, Watchdog anomalies, incidents, alerts, and more. It also expands on issues with upstream and downstream dependencies. This feature works best if your APM services are tagged by **team** and **service**.

{{< whatsnext desc="Additional documentation:">}}
    {{< nextlink href="bits_ai/getting_started/" >}}Getting Started{{< /nextlink >}}
    {{< nextlink href="bits_ai/query_examples/" >}}Example Natural Language Queries{{< /nextlink >}}
{{< /whatsnext >}}

### Streamline incident management

Within the [Datadog Incident Management][2] product, Bits AI can help you 

> - **Stay informed:** Query incident details, such as "Give me a few bullet points on incident 2781." Ask Datadog to send you an update on the incident in 15 minutes.

Reviewer: ```suggestion
- **Stay informed:** Ask Datadog to send you an update on the incident in 15 minutes.
```
